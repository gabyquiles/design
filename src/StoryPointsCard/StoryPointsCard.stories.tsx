import * as React from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import StoryPointsCard from './StoryPointsCard';

function StoryPointsCardWithValue({value}: {value: string}) {
    return (
        <StoryPointsCard>
            {value}
        </StoryPointsCard>
    )
}

const meta: Meta = {
    title: 'StoryPointsCard',
    component: StoryPointsCardWithValue,
}

export const FirstStory: StoryObj = {
    args: {
        value: '5'
    }
}

export default meta;
