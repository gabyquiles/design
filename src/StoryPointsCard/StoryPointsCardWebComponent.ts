// import { WebComponentProps } from '@lit/react';
import {html, css, LitElement, type PropertyDeclarations} from 'lit';
import {customElement, property} from 'lit/decorators.js';

@customElement('storypoints-card')
export class StoryPointsCard extends LitElement {
  static styles = css`
    .card {
      background-color: rgb(255, 255, 255);
      color: rgb(0, 0, 0);
      transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
      border-radius: 4px;
      box-shadow: rgba(0, 0, 0, 0.2) 0px 2px 1px -1px, rgba(0, 0, 0, 0.14) 0px 1px 1px 0px, rgba(0, 0, 0, 0.12) 0px 1px 3px 0px;
      background-image: linear-gradient(rgba(255, 255, 255, 0.05), rgba(255, 255, 255, 0.05));
      overflow: hidden;
      width: 6.6em;
      height: 9.2em;
      margin: 1em;
      display: flex;
      justify-content: center;
      align-items: center;
    }
    .card span {
      font-size: 2em;
    }
  `;

  @property() value = '';

  render() {
      return html`<div class="card"><span>${this.value}</span></div>`;
  }
}