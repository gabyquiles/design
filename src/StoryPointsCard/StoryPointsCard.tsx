import * as React from 'react';
import {createComponent} from '@lit/react';
import { StoryPointsCard as StoryPointsCardWC } from './StoryPointsCardWebComponent';

export const StoryPointsCard = createComponent<StoryPointsCardWC>({
    react: React,
    tagName: 'storypoints-card',
    elementClass: StoryPointsCardWC,
    displayName: 'StoryPointsCard'
})

interface StoryPointsCardProps extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement> {
    value: string
}

declare global {
    namespace JSX {
      interface IntrinsicElements {
        'storypoints-card': StoryPointsCardProps
      }
    }
}

export default StoryPointsCard;