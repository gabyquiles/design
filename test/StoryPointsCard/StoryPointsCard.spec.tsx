import * as React from 'react';
import { render, waitFor } from '@testing-library/react';
import { screen } from 'shadow-dom-testing-library';

import StoryPointsCard from "../../src/StoryPointsCard/StoryPointsCard"

describe('StoryPointsCard', () => {
    it('renders correctly', () => {
        const { container } = render(<StoryPointsCard>7</StoryPointsCard>)
        expect(container).toMatchSnapshot();
    })
    it('shows the storypoints value', async () => {
        render(<StoryPointsCard>7</StoryPointsCard>)
        expect(screen.getByShadowText('7')).toBeInTheDocument()
    })
})