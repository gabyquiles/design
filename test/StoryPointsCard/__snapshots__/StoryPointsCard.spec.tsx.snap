// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StoryPointsCard > renders correctly 1`] = `
<div>
  <storypoints-card>
    7
  </storypoints-card>
</div>
`;
